src/adagrams.js

```diff
@@ -1,8 +1,162 @@
-const Adagrams = {
-  drawLetters() {
-    // Implement this method for wave 1
-  },
-};
+class Adagrams {
+  static generatePool() {
+    const lettersCounts = {
+      A: 9,
+      B: 2,
+      C: 2,
+      D: 4,
+      E: 12,
+      F: 2,
+      G: 3,
+      H: 2,
+      I: 9,
+      J: 1,
+      K: 1,
+      L: 4,
+      M: 2,
+      N: 6,
+      O: 8,
+      P: 2,
+      Q: 1,
+      R: 6,
+      S: 4,
+      T: 6,
+      U: 4,
+      V: 2,
+      W: 2,
+      X: 1,
+      Y: 2,
+      Z: 1,
+    };
+
+    let output = [];
+    for (let key in lettersCounts) {
+      const value = lettersCounts[key];
+      for (let i = 0; i < value; i++) {
+        output.push(key);
+      }
+    }
+    return output;
+  }
+
+  static getRandomIndices(numberOfLetters, maxIndex) {
```

Comment:
Since the method is more returning a set of random **elements** as opposed to random indices, I'd suggest naming this method `getRandomElements`